pkg/agent/flowexporter/exporter/exporter_perf_test.go

```diff
@@ -0,0 +1,245 @@
+			TCPState:                   "SYN_SENT",
+		}
+		connections[i] = conn
+	}
+	return connections
+}
+
+func getUpdateConnections(b *testing.B, conns []*flowexporter.Connection, numOfNewConn int, numOfDyingConn int, index int) []*flowexporter.Connection {
+	currentTime := time.Now().Add(time.Duration(index) * time.Second)
+	connections := make([]*flowexporter.Connection, 0)
+
+	// update connections and delete connections in dying state
+	for i := 0; i < len(conns); i++ {
+		if conns[i].TCPState == "TIME_WAIT" {
+			continue
+		}
+		conn := *conns[i]
+		conn.OriginalPackets += 10
+		conn.OriginalBytes += 10
+		conn.ReversePackets += 5
+		conn.ReverseBytes += 5
+		randomNum := getRandomNum(b, 1000)
+		conn.StopTime = currentTime.Add(-time.Duration(randomNum) * time.Millisecond)
+		connections = append(connections, &conn)
+	}
+
+	// add new connections
+	for i := 0; i < numOfNewConn; i++ {
+		randomNum1 := getRandomNum(b, 255)
+		randomNum2 := getRandomNum(b, 255)
+		randomNum3 := getRandomNum(b, 255)
+		src := net.ParseIP(fmt.Sprintf("192.%d.%d.%d", randomNum1, randomNum2, randomNum3))
+		dst := net.ParseIP(fmt.Sprintf("192.%d.%d.%d", randomNum1, randomNum2, randomNum3))
+		randomPort := getRandomNum(b, 90000)
+		flowKey := flowexporter.Tuple{SourceAddress: src, DestinationAddress: dst, Protocol: 6, SourcePort: uint16(randomPort), DestinationPort: uint16(randomPort)}
+		conn := &flowexporter.Connection{
+			StartTime:                  currentTime.Add(-time.Duration(randomNum1) * time.Millisecond),
+			StopTime:                   currentTime.Add(-time.Duration(randomNum1) * time.Millisecond),
+			LastExportTime:             currentTime.Add(-time.Duration(randomNum1) * time.Millisecond),
+			IsPresent:                  true,
+			DoneExport:                 false,
+			FlowKey:                    flowKey,
+			OriginalPackets:            100,
+			OriginalBytes:              10,
+			ReversePackets:             50,
+			ReverseBytes:               5,
+			SourcePodNamespace:         "ns1",
+			SourcePodName:              "pod1",
+			DestinationPodNamespace:    "ns2",
+			DestinationPodName:         "pod2",
+			DestinationServicePortName: "service",
+			DestinationServiceAddress:  net.ParseIP("0.0.0.0"),
+			TCPState:                   "SYN_SENT",
+		}
+		connections = append(connections, conn)
+	}
+
+	// change connections to dying state, which will be deleted in next round
+	randomNum := int(getRandomNum(b, int64(len(connections)-numOfDyingConn)))
+	for i := randomNum; i < numOfDyingConn+randomNum; i++ {
+		connections[i].TCPState = "TIME_WAIT"
+	}
+	return connections
+}
+
+func statMaxMemAlloc(maxAlloc *uint64, interval time.Duration, stopCh chan struct{}) {
+	var memStats goruntime.MemStats
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			goruntime.ReadMemStats(&memStats)
+			if memStats.Alloc > *maxAlloc {
+				*maxAlloc = memStats.Alloc
+			}
+		case <-stopCh:
+			return
+		}
+	}
+}
```

Comment:
do we still need this memstat collection? Are these something different from what we are getting from benchmark test?

Reply:
No we don't need this any more. Have it removed. Thanks!